src/lib/components/chat/Messages.svelte

```diff
@@ -221,6 +222,19 @@
 			scrollToBottom();
 		}, 100);
 	};
+
+	const deleteMessagePair = async (messageId) => {
+		history.messages[messageId].deleted = true;
+		history.messages[history.messages[messageId].childrenIds[0]].deleted = true;
```

Comment:
Thanks for the PR! I believe this should apply to all ancestor messages recursively, not just the first one.

Reply:
> Thanks for the PR! I believe this should apply to all ancestor messages recursively, not just the first one.

Do you mean we should also be removing all subsequent message/response pairs after the deleted message as well?

Reply:
I think got a bit confused as I was leaving the comment, I meant `delete` flag should be set to true for all children messages, right now this won't delete any of the regenerated responses.

Reply:
Also, another thing, it might be better to directly manipulate the history graph object instead of flagging the messages but this should do the job for now!

Reply:
Updated to set `delete=true` flag to all children messages recursively